lib/sycamore/sycamore/connectors/weaviate/weaviate_scan.py

```diff
@@ -0,0 +1,26 @@
+from dataclasses import asdict
+from sycamore.connectors.weaviate.weaviate_writer import WeaviateClientParams
+from sycamore.plan_nodes import Scan
+from sycamore.data import Document
+from weaviate import WeaviateClient
+from ray.data import Dataset, from_items
+
+
+class WeaviateScan(Scan):
+    def __init__(self, collection_name: str, connection_params: WeaviateClientParams, **kwargs):
+        super().__init__(**kwargs)
+        self._collection_name = collection_name
+        self._connection_params = connection_params
+
+    def execute(self) -> Dataset:
+        documents = []
+        with WeaviateClient(**asdict(self._connection_params)) as wcl:
```

Comment:
Similar to the test case, would suggest checking whether the client starts in <timeout seconds

Reply:
The test case does that bc it's using embedded weaviate - so it has to start it up and potentially download a binary etc. 
I'm pretty sure I can assume I have a running weaviate instance when I try to read from it. Also you can add a timeout to the connection params, like [here](https://github.com/aryn-ai/sycamore/blob/bd05d08086711ac54306c8777f13e12161c0fca6/examples/simple_weaviate.py#L36)

Reply:
Makes sense, discussed offline as well